mountpoint-s3/src/prefetch.rs

```diff
@@ -198,8 +204,11 @@ where
                 return Ok(part_bytes);
             }
 
-            response.extend_from_slice(&part_bytes[..]);
-            to_read -= part_bytes.len() as u64;
+            let part_len = part_bytes.len() as u64;
+            if response.extend(part_bytes).is_err() {
+                return Err(PrefetchReadError::Integrity);
```

Comment:
Do we need to do the same cancel inflight tasks work here as we do when `current_task.read()` above fails?